Sources/Host/Darwin/Mach.cpp

```diff
@@ -80,14 +80,69 @@ ErrorCode Mach::readMemory(ProcessThreadId const &ptid, Address const &address,
   if (kret != KERN_SUCCESS)
     return kErrorInvalidAddress;
 
-  *count = curr_bytes_read;
+  if (count)
+    *count = curr_bytes_read;
 
   return kSuccess;
 }
 
 ErrorCode Mach::writeMemory(ProcessThreadId const &ptid, Address const &address,
                             void const *buffer, size_t length, size_t *count) {
-  return kErrorUnsupported;
+  /*
+   * We want to write in the process memory, but most of the write will be
```

Comment:
The mach debugging APIs do not allow writing to pages mapped without write permissions; remap them before writing.